uaclient/cli.py

```diff
@@ -1009,6 +1037,77 @@ def action_attach(args, *, cfg):
     )
 
 
+def _write_command_output_to_file(
+    cmd, filename: str, return_codes: List[int] = None
+) -> None:
+    """Helper which runs a command and writes output or error to filename."""
+    try:
+        out, _ = util.subp(cmd.split(), rcs=return_codes)
+    except util.ProcessExecutionError as e:
+        util.write_file(filename, str(e))
+    else:
+        util.write_file(filename, out)
+
+
+def _collect_file(path: str, out_dir: str) -> None:
+    if os.path.isfile(path):
+        shutil.copy(path, out_dir)
+
+
+# We have to assert root here, because the logs are not non-root user readable
+@assert_root
+def action_collect_logs(args, *, cfg: config.UAConfig):
+    current_datetime = datetime.now().strftime("%b-%d-%H%M")
+
+    output_file = args.output or UA_COLLECT_LOGS_FILE
+    output_dir = "/tmp/ua-logs-{}".format(current_datetime)
+    if os.path.exists(output_dir):
+        shutil.rmtree(output_dir)
+    os.makedirs(output_dir)
+
+    _write_command_output_to_file(
+        "cloud-id", "{}/cloud-id.txt".format(output_dir)
+    )
+    _write_command_output_to_file(
+        "ua status --format json", "{}/ua-status.json".format(output_dir)
+    )
+    _write_command_output_to_file(
+        "canonical-livepatch status",
+        "{}/livepatch-status.txt".format(output_dir),
+    )
+    _write_command_output_to_file(
+        "systemctl list-timers --all",
+        "{}/systemd-timers.txt".format(output_dir),
+    )
+    _write_command_output_to_file(
+        "journalctl --boot=0 -o short-precise",
+        "{}/journalctl.txt".format(output_dir),
+    )
+    _write_command_output_to_file("dmesg", "{}/dmesg.txt".format(output_dir))
+
+    for service in UA_SERVICES:
+        _write_command_output_to_file(
+            "systemctl status {}".format(service),
+            "{}/{}.txt".format(output_dir, service),
+            return_codes=[0, 3],
+        )
+
+    ua_logs = (
+        "/etc/ubuntu-advantage/uaclient.conf",
```

Comment:
This file can also be overridden by the user. For example, users can use a local config file to drive UA behavior